x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/custom_response_actions_rule.ts

```diff
@@ -0,0 +1,142 @@
+import type { ITelemetryReceiver } from '../receiver';
+import type { ResponseActionRules, ResponseActionsRuleResponseAggregations } from '../types';
+import type { TaskExecutionPeriod } from '../task';
+import type { ITaskMetricsService } from '../task_metrics.types';
+
+export function createTelemetryCustomResponseActionRulesTaskConfig(maxTelemetryBatch: number) {
+  const taskName = 'Security Solution Response Actions Rules Telemetry';
+  const taskType = 'security:telemetry-response-actions-rules';
+  return {
+    type: taskType,
+    title: taskName,
+    interval: '24h',
+    timeout: '10m',
+    version: '1.0.0',
+    runTask: async (
+      taskId: string,
+      logger: Logger,
+      receiver: ITelemetryReceiver,
+      sender: ITelemetryEventsSender,
+      taskMetricsService: ITaskMetricsService,
+      taskExecutionPeriod: TaskExecutionPeriod
+    ) => {
+      const mdc = { task_id: taskId, task_execution_period: taskExecutionPeriod };
+      const log = newTelemetryLogger(logger.get('response_actions_rules'), mdc);
+      const usageCollector = sender.getTelemetryUsageCluster();
+      const usageLabelEndpointPrefix: string[] = [
+        'security_telemetry',
+        'endpoint-response-actions-rules',
+      ];
+      const usageLabelOsqueryPrefix: string[] = [
+        'security_telemetry',
+        'osquery-response-actions-rules',
+      ];
+      const trace = taskMetricsService.start(taskType);
+
+      log.l('Running response actions rules telemetry task');
+
+      try {
+        const [clusterInfoPromise, licenseInfoPromise] = await Promise.allSettled([
+          receiver.fetchClusterInfo(),
+          receiver.fetchLicenseInfo(),
+        ]);
+
+        const clusterInfo = safeValue(clusterInfoPromise);
+        const licenseInfo = safeValue(licenseInfoPromise);
+
+        const {
+          body: { aggregations },
+        } = (await receiver.fetchResponseActionsRules()) ?? {};
+
+        if (!aggregations || !aggregations.actionTypes) {
+          log.debug('no custom response action rules found');
+          await taskMetricsService.end(trace);
+          return 0;
+        }
+
+        const responseActionRules = (
+          aggregations as unknown as ResponseActionsRuleResponseAggregations
+        ).actionTypes.buckets.reduce<ResponseActionRules>((acc, agg) => {
+          if (agg.key === '.endpoint') {
+            acc.endpoint = agg.rulesInfo.buckets.map((rule) => rule.key);
+          } else if (agg.key === '.osquery') {
+            acc.osquery = agg.rulesInfo.buckets.map((rule) => rule.key);
+          }
+          return acc;
+        }, {} as ResponseActionRules);
+
+        const shouldNotProcessTelemetry =
+          responseActionRules.endpoint === undefined ||
+          responseActionRules.osquery === undefined ||
+          responseActionRules.endpoint.length === 0 ||
+          responseActionRules.osquery.length === 0;
+
+        if (shouldNotProcessTelemetry) {
+          return 0;
```

Comment:
don't forget to call `await taskMetricsService.end(trace);`

Reply:
done 2cd4c87c9694e445b6dcbd0e6cb4c4099677dfb7